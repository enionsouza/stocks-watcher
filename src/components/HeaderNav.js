import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import { BsChevronLeft } from 'react-icons/bs';
import { FaMicrophone } from 'react-icons/fa';
import { IoIosSettings } from 'react-icons/io';

const HeaderNav = ({ type, symbol }) => {
  const title = {
    home: 'Stocks Watcher',
    details: 'Details',
    noMatch: 'Error',
  };

  return (
    <div className="d-flex justify-content-around align-items-center py-1 dark-bg position-sticky">
      <Link to="/" className="link-button mx-3 fs-5">
        {type !== 'home' ? <BsChevronLeft /> : ''}
      </Link>
      <h3 className="body-title-text flex-grow-1 text-center mb-0 pt-1">{`${title[type]}${symbol ? ` - ${symbol}` : ''}`}</h3>
      <button
        type="button"
        onClick={() => {}}
        className="link-button fs-5"
      >
        <FaMicrophone />
      </button>
      <button
        type="button"
        onClick={() => {}}
        className="link-button mx-3 fs-5"
      >
        <IoIosSettings />
      </button>
    </div>
  );
};

HeaderNav.propTypes = {
  type: PropTypes.string.isRequired,
  // eslint-disable-next-line react/require-default-props
  symbol: PropTypes.string,
};

HeaderNav.default = {
  type: 'home',
};

export default HeaderNav;
