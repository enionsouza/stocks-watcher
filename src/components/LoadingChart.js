const LoadingChart = () => (
  <div style={{ height: '90vh' }} className="d-flex justify-content-center align-items-center">
    <h3>Loading data...</h3>
  </div>
);

export default LoadingChart;
