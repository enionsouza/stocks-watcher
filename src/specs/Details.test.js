import { Provider } from 'react-redux';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { render } from '@testing-library/react';
import store from '../redux/configureStore';
import HeaderNav from '../components/HeaderNav';
import LoadingChart from '../components/LoadingChart';

describe('Details page', () => {
  jest.setTimeout(300000);
  describe('Use Jest snapshots to test the Details page', () => {
    it('renders Details page \'Loading...\' message while waiting for the necessary data', async () => {
      const detailsPage = render(
        <Provider store={store}>
          <Router>
            <Switch>
              <Route exact path="/">
                <HeaderNav type="details" symbol="AAPL" />
                <LoadingChart />
              </Route>
            </Switch>
          </Router>
        </Provider>,
      );

      expect(detailsPage.asFragment()).toMatchSnapshot();
    });
  });
});
