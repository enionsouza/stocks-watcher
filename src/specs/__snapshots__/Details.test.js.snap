// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details page Use Jest snapshots to test the Details page renders Details page 'Loading...' message while waiting for the necessary data 1`] = `
<DocumentFragment>
  <div
    class="d-flex justify-content-around align-items-center py-1 dark-bg position-sticky"
  >
    <a
      class="link-button mx-3 fs-5"
      href="/"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.354 1.646a.5.5 0 0 1 0 .708L5.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0z"
          fill-rule="evenodd"
        />
      </svg>
    </a>
    <h3
      class="body-title-text flex-grow-1 text-center mb-0 pt-1"
    >
      Details - AAPL
    </h3>
    <button
      class="link-button fs-5"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 640 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M633.82 458.1l-157.8-121.96C488.61 312.13 496 285.01 496 256v-48c0-8.84-7.16-16-16-16h-16c-8.84 0-16 7.16-16 16v48c0 17.92-3.96 34.8-10.72 50.2l-26.55-20.52c3.1-9.4 5.28-19.22 5.28-29.67V96c0-53.02-42.98-96-96-96s-96 42.98-96 96v45.36L45.47 3.37C38.49-2.05 28.43-.8 23.01 6.18L3.37 31.45C-2.05 38.42-.8 48.47 6.18 53.9l588.36 454.73c6.98 5.43 17.03 4.17 22.46-2.81l19.64-25.27c5.41-6.97 4.16-17.02-2.82-22.45zM400 464h-56v-33.77c11.66-1.6 22.85-4.54 33.67-8.31l-50.11-38.73c-6.71.4-13.41.87-20.35.2-55.85-5.45-98.74-48.63-111.18-101.85L144 241.31v6.85c0 89.64 63.97 169.55 152 181.69V464h-56c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16z"
        />
      </svg>
    </button>
    <button
      class="link-button mx-3 fs-5"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 6a3.939 3.939 0 0 0-3.934 3.934h2C10.066 8.867 10.934 8 12 8s1.934.867 1.934 1.934c0 .598-.481 1.032-1.216 1.626a9.208 9.208 0 0 0-.691.599c-.998.997-1.027 2.056-1.027 2.174V15h2l-.001-.633c.001-.016.033-.386.441-.793.15-.15.339-.3.535-.458.779-.631 1.958-1.584 1.958-3.182A3.937 3.937 0 0 0 12 6zm-1 10h2v2h-2z"
        />
        <path
          d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
        />
      </svg>
    </button>
  </div>
  <div
    class="d-flex justify-content-center align-items-center"
    style="height: 90vh;"
  >
    <h3>
      Loading data...
    </h3>
  </div>
</DocumentFragment>
`;
